mono/mini/mini-runtime.h

```diff
@@ -337,6 +337,29 @@ extern GList* mono_aot_paths;
 extern MonoDebugOptions mini_debug_options;
 extern GSList *mono_interp_only_classes;
 
+/*
+This struct describes what execution engine feature to use.
+This subsume, and will eventually sunset, mono_aot_only / mono_llvm_only and friends.
+The goal is to transition us to a place were we can more easily compose/describe what features we need for a given execution mode.
+
+A good feature flag is checked alone, a bad one described many things and keeps breaking some of the modes
+*/
+typedef struct {
+	/*
+	 * If true, trampolines are to be fetched from the AOT runtime instead of JIT compiled
+	 */
+	gboolean use_aot_trampolines;
+} MonoEEFeatures;
+
+extern MonoEEFeatures mono_ee_features;
+
+//XXX this enum *MUST extend MonoAotMode as they are consumed together.
+typedef enum {
+	/* Always execute with interp, will use JIT to produce trampolines */
+	MONO_EE_MODE_INTERP = MONO_AOT_MODE_LAST,
+} MonoEEMode;
```

Comment:
This looks a bit whacky. Is `MONO_EE_MODE_INTERP` something we wanna expose via the API (like we do for `MonoAotMode`)? Can't we just add it there?

If you wanna keep it like that, you should define `MONO_AOT_MODE_LAST = 1000` or something, so we have some room to come up with some other crazy execution modes 😅

Reply:
I don't think we should expose mono_runtime_set_execution_mode/MonoEEMode until needed.
We `believe` this is a good solution. So let's first see it through and then think on what to do with the public API.

But I see your point, my design will bomb if you compile with newer headers and try to run against older runtimes. Fixed it.